# Linear Regression Assignment

In [1]:
#Importing necessary libraries
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import warnings
warnings.filterwarnings('ignore')

#Import necessary Linear Regression model 
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score,mean_squared_error,mean_absolute_error

1.Load the dataset using pandas

In [3]:
#import data
df=pd.read_csv("data.csv")
df.head()

YearsExperience   Salary
0              1.1  39343.0
1              1.3  46205.0
2              1.5  37731.0
3              2.0  43525.0
4              2.2  39891.0

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 2 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   YearsExperience  30 non-null     float64
 1   Salary           30 non-null     float64
dtypes: float64(2)
memory usage: 608.0 bytes


In [12]:
df.columns

Index(['YearsExperience', 'Salary'], dtype='object')

In [15]:
df.describe().T

count          mean           std      min       25%  \
YearsExperience   30.0      5.313333      2.837888      1.1      3.20   
Salary            30.0  76003.000000  27414.429785  37731.0  56720.75   

                     50%        75%       max  
YearsExperience      4.7       7.70      10.5  
Salary           65237.0  100544.75  122391.0

In [17]:
df.dtypes

YearsExperience    float64
Salary             float64
dtype: object

2.Extract data fromYearsExperience column is a variable named X

In [4]:
X=pd.DataFrame(df['YearsExperience'])
X

YearsExperience
0               1.1
1               1.3
2               1.5
3               2.0
4               2.2
5               2.9
6               3.0
7               3.2
8               3.2
9               3.7
10              3.9
11              4.0
12              4.0
13              4.1
14              4.5
15              4.9
16              5.1
17              5.3
18              5.9
19              6.0
20              6.8
21              7.1
22              7.9
23              8.2
24              8.7
25              9.0
26              9.5
27              9.6
28             10.3
29             10.5

3.Extract data from salary column is a variable named Y

In [23]:
Y=pd.DataFrame(df['Salary'])
Y

Salary
0    39343.0
1    46205.0
2    37731.0
3    43525.0
4    39891.0
5    56642.0
6    60150.0
7    54445.0
8    64445.0
9    57189.0
10   63218.0
11   55794.0
12   56957.0
13   57081.0
14   61111.0
15   67938.0
16   66029.0
17   83088.0
18   81363.0
19   93940.0
20   91738.0
21   98273.0
22  101302.0
23  113812.0
24  109431.0
25  105582.0
26  116969.0
27  112635.0
28  122391.0
29  121872.0

In [ ]:
4. Divide the dataset into two parts for training and testing in 66% and 33% proportion

In [25]:
X_train,x_test,Y_train,y_test=train_test_split(X,Y,test_size=0.33,random_state=42)

5. Create and train LinearRegression Model on training set

In [26]:
lr=LinearRegression()

In [27]:
lr.fit(X_train,Y_train)

LinearRegression()

6. Make predictions based on the testing set using the trained model

In [29]:
y_prediction=lr.predict(x_test)

In [30]:
y_prediction

array([[115814.30756236],
       [ 71511.92534771],
       [102617.85328566],
       [ 75282.34085534],
       [ 55487.65944028],
       [ 60200.67882482],
       [122412.53470072],
       [107330.8726702 ],
       [ 63028.49045554],
       [ 35692.97802523]])

In [34]:
y_prediction=pd.DataFrame(y_prediction,columns=['predicted'])
y_prediction

predicted
0  115814.307562
1   71511.925348
2  102617.853286
3   75282.340855
4   55487.659440
5   60200.678825
6  122412.534701
7  107330.872670
8   63028.490456
9   35692.978025

7. Check the performance by calculating the r2 score of the model

In [46]:
from sklearn import metrics
print(f'Mean squared Error:',metrics.mean_squared_error(y_test,y_prediction))
print(f'Mean Absolute Error:',metrics.mean_absolute_error(y_test,y_prediction))
print(f'Root Mean Squared Error:',np.sqrt(mean_squared_error(y_test,y_prediction)))

Mean squared Error: 35301898.887134895
Mean Absolute Error: 4956.5232614449005
Root Mean Squared Error: 5941.540110706558


In [51]:
r2_score(y_test,y_prediction)

0.9553063138044949

R-Squared value of 0.955 would indicate that 95% of the variance of the dependent variable(Y) is explained by the variance of the independent variable(X).
The above showing High r2_score which is indicated a goodness of fit.

# Logistic Regression Assignment

1. Load the dataset using pandas

In [5]:
df1=pd.read_csv("diabetes.csv")
df1.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [54]:
df1.columns

Index(['Pregnancies', 'Glucose', 'BloodPressure', 'SkinThickness', 'Insulin',
       'BMI', 'DiabetesPedigreeFunction', 'Age', 'Outcome'],
      dtype='object')

In [55]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [57]:
df1.shape

(768, 9)

In [61]:
df1.isnull().sum()   #No null Values

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

In [62]:
df1.duplicated()   #No duplicates

0      False
1      False
2      False
3      False
4      False
       ...  
763    False
764    False
765    False
766    False
767    False
Length: 768, dtype: bool

In [63]:
df1.columns

Index(['Pregnancies', 'Glucose', 'BloodPressure', 'SkinThickness', 'Insulin',
       'BMI', 'DiabetesPedigreeFunction', 'Age', 'Outcome'],
      dtype='object')

2. Extract data fromOutcome column is a variable named Y

In [65]:
Y=pd.DataFrame(df1['Outcome'])
Y

Outcome
0          1
1          0
2          1
3          0
4          1
..       ...
763        0
764        0
765        0
766        1
767        0

[768 rows x 1 columns]

3. Extract data from every column except Outcome column in a variable named X

In [66]:
X=pd.DataFrame(df1.iloc[:,:-1])
X

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  
0                       0.627   50  
1                       0.351   31  
2                       0.672   32  
3                       0.167   21  
4                       2.288   33  
..                        ...  ...  
763                     0.171   63  
764                     0.340   27  
765                     0.245   30  
766                     0.349   47  
767                     0.315   23  

[768 rows x 8 columns]

4. Divide the dataset into two parts for training and testing in 70% and 30% proportion

In [83]:
X_train,x_test,Y_train,y_test=train_test_split(X,Y,test_size=0.33,random_state=42)

In [84]:
X_train

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
464           10      115             98              0        0  24.0   
223            7      142             60             33      190  28.8   
393            4      116             72             12       87  22.1   
766            1      126             60              0        0  30.1   
570            3       78             70              0        0  32.5   
..           ...      ...            ...            ...      ...   ...   
71             5      139             64             35      140  28.6   
106            1       96            122              0        0  22.4   
270           10      101             86             37        0  45.6   
435            0      141              0              0        0  42.4   
102            0      125             96              0        0  22.5   

     DiabetesPedigreeFunction  Age  
464                     1.022   34  
223                     0.687   61  
393                     0.463   37  
766                     0.349   47  
570                     0.270   39  
..                        ...  ...  
71                      0.411   26  
106                     0.207   27  
270                     1.136   38  
435                     0.205   29  
102                     0.262   21  

[514 rows x 8 columns]

5. Create and train Logistic Regression Model on training set

In [85]:
from sklearn.linear_model import LogisticRegression
log_reg=LogisticRegression()

In [86]:
log_reg.fit(X_train,Y_train)

LogisticRegression()

6. Make predictions based on the testing set using the trained model

In [87]:
y_pred=log_reg.predict(x_test)

In [89]:
y_pred

array([0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0,
       0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 1, 1, 0,
       0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1,
       0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0,
       0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0,
       0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1,
       0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 1, 0,
       0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0,
       0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1,
       1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0,
       1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0], dtype=int64)

In [ ]:
7. Check the performance by calculating the confusion matrix and accuracy score of the model

In [92]:
from sklearn.metrics import confusion_matrix,accuracy_score
print(f'Accuracy score of the Logisctic Regression model is :',accuracy_score(y_test,y_pred))

Accuracy score of the Logisctic Regression model is : 0.7480314960629921


In [96]:
Confusion_Matrix=confusion_matrix(y_test,y_pred)
print(Confusion_Matrix)

[[136  32]
 [ 32  54]]


In [95]:
(136+54)/(136+32+32+54)

0.7480314960629921

# Decision Tree Assignment

In [ ]:
1. Load the dataset using pandas

In [6]:
df2=pd.read_csv("diabetes.csv")
df2.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

2. Extract data fromOutcome column is a variable named Y

In [99]:
Y=pd.DataFrame(df2['Outcome'])
Y

Outcome
0          1
1          0
2          1
3          0
4          1
..       ...
763        0
764        0
765        0
766        1
767        0

[768 rows x 1 columns]

3. Extract data from every column except Outcome column in a variable named X

In [100]:
X=pd.DataFrame(df2.iloc[:,:-1])
X

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  
0                       0.627   50  
1                       0.351   31  
2                       0.672   32  
3                       0.167   21  
4                       2.288   33  
..                        ...  ...  
763                     0.171   63  
764                     0.340   27  
765                     0.245   30  
766                     0.349   47  
767                     0.315   23  

[768 rows x 8 columns]

4. Divide the dataset into two parts for training and testing in 70% and 30% proportion

In [102]:
X_train,x_test,Y_train,y_test=train_test_split(X,Y,test_size=0.3,random_state=42)

In [104]:
print(X_train.shape)
print(x_test.shape)
print(Y_train.shape)
print(y_test.shape)

(537, 8)
(231, 8)
(537, 1)
(231, 1)


5. Create and train Decision Tree Model on training set

In [105]:
from sklearn.tree import DecisionTreeClassifier

In [106]:
cl=DecisionTreeClassifier()

In [107]:
cl.fit(X_train,Y_train)

DecisionTreeClassifier()

6. Make predictions based on the testing set using the trained model

In [109]:
y_train_pred=cl.predict(X_train)
y_test_pred=cl.predict(x_test)

In [110]:
y_train_pred

array([0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1,
       0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0,
       0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0,
       1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0,
       0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1,
       0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1,
       0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0,
       0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0,
       1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 1, 0,
       0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1, 1, 1,
       1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 1,
       1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, 0,
       0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1,

In [111]:
y_test_pred

array([0, 1, 0, 0, 0, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0,
       0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 1, 1,
       0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 1,
       0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 1, 1,
       0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1,
       0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 1,
       0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0,
       0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 1,
       1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0,
       0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1,
       1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0], dtype=int64)

7. Check the performance by calculating the confusion matrix and accuracy score of the model

In [114]:
print(f'Train score {accuracy_score(y_train_pred,Y_train)}')
print(f'Test score {accuracy_score(y_test_pred,y_test)}')

Train score 1.0
Test score 0.696969696969697


In [115]:
print(f'Accuracy score of the Decision Tree model is :',accuracy_score(y_test,y_test_pred))

Accuracy score of the Decision Tree model is : 0.696969696969697


In [116]:
confusion_matrix(y_test_pred,y_test)

array([[109,  28],
       [ 42,  52]], dtype=int64)

In [118]:
from sklearn.metrics import classification_report

In [119]:
print(classification_report(y_test_pred,y_test))

              precision    recall  f1-score   support

           0       0.72      0.80      0.76       137
           1       0.65      0.55      0.60        94

    accuracy                           0.70       231
   macro avg       0.69      0.67      0.68       231
weighted avg       0.69      0.70      0.69       231



# Random Forest Assignment

1. Load the dataset using pandas

In [120]:
df3=pd.read_csv("C://Users//ysadmin//Desktop//diabetes.csv")
df3.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

2. Extract data fromOutcome column is a variable named Y

In [121]:
Y=pd.DataFrame(df3['Outcome'])
Y

Outcome
0          1
1          0
2          1
3          0
4          1
..       ...
763        0
764        0
765        0
766        1
767        0

[768 rows x 1 columns]

3. Extract data from every column except Outcome column in a variable named X.

In [122]:
X=pd.DataFrame(df3.iloc[:,:-1])
X

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  
0                       0.627   50  
1                       0.351   31  
2                       0.672   32  
3                       0.167   21  
4                       2.288   33  
..                        ...  ...  
763                     0.171   63  
764                     0.340   27  
765                     0.245   30  
766                     0.349   47  
767                     0.315   23  

[768 rows x 8 columns]

4. Divide the dataset into two parts for training and testing in 70% and 30% proportion

In [123]:
X_train,x_test,Y_train,y_test=train_test_split(X,Y,train_size=0.7,random_state=42)

5. Create and train Random Forest Model on training set

In [124]:
from sklearn.ensemble import RandomForestClassifier

In [132]:
cl1=RandomForestClassifier(n_estimators=600)
cl1.fit(X_train,Y_train)

RandomForestClassifier(n_estimators=600)

6. Make predictions based on the testing set using the trained model

In [133]:
y_test_pred=cl1.predict(x_test)

In [134]:
y_test

Outcome
668        0
324        0
624        0
690        0
473        0
..       ...
619        1
198        1
538        0
329        0
302        0

[231 rows x 1 columns]

7. Check the performance by calculating the confusion matrix and accuracy score of the model

In [135]:
Confusion_Matrix=confusion_matrix(y_test,y_test_pred)
Confusion_Matrix

array([[123,  28],
       [ 29,  51]], dtype=int64)

In [136]:
Accuracy_Score=accuracy_score(y_test,y_test_pred)
print(Accuracy_Score)

0.7532467532467533


# Naïve-Bayes Assignment

1. Load the dataset using pandas

In [138]:
df4=pd.read_csv("C://Users//ysadmin//Desktop//diabetes.csv")
df4.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

2. Extract data fromOutcome column is a variable named Y

In [139]:
Y=pd.DataFrame(df4['Outcome'])
Y

Outcome
0          1
1          0
2          1
3          0
4          1
..       ...
763        0
764        0
765        0
766        1
767        0

[768 rows x 1 columns]

3. Extract data from every column except Outcome column in a variable named X

In [140]:
X=pd.DataFrame(df4.iloc[:,:-1])
X

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  
0                       0.627   50  
1                       0.351   31  
2                       0.672   32  
3                       0.167   21  
4                       2.288   33  
..                        ...  ...  
763                     0.171   63  
764                     0.340   27  
765                     0.245   30  
766                     0.349   47  
767                     0.315   23  

[768 rows x 8 columns]

4. Divide the dataset into two parts for training and testing in 70% and 30% proportion

In [141]:
X_train,x_test,Y_train,y_test=train_test_split(X,Y,train_size=0.7,random_state=42)

5. Create and train Naïve Bayes Model on training set

In [143]:
from sklearn.naive_bayes import GaussianNB

In [144]:
nb=GaussianNB()
nb.fit(X_train,Y_train)

GaussianNB()

6. Make predictions based on the testing set using the trained model

In [145]:
y_pred=nb.predict(x_test)

In [146]:
y_pred

array([0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0,
       1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1,
       0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 1,
       0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0,
       0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 1,
       0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 1,
       0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0,
       0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 1, 0,
       0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0,
       0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1,
       1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=int64)

7. Check the performance by calculating the confusion matrix and accuracy score of the model

In [147]:
Confusion_Matrix=confusion_matrix(y_test,y_pred)
print(Confusion_Matrix)

[[119  32]
 [ 27  53]]


In [149]:
print(f'The accuracy score of naive bayes model is:',  metrics.accuracy_score(y_test, y_pred)*100)

The accuracy score of naive bayes model is: 74.45887445887446
